use-timescale/migration/dual-write-and-backfill.md

```diff
@@ -157,9 +185,101 @@ commit;
+
+```
+pg_dumpall -d "$SOURCE" \
+  --quote-all-identifiers \
+  --roles-only \
+  --file=roles.sql
+```
+
+Timescale services do not support roles with superuser access. If your SQL
+dump includes roles that have such permissions, you'll need to modify the file
+to be compliant with the security model.
+
+You can use the following `sed` command to remove unsupported statements and
+permissions from your roles.sql file:
+
+```
+sed -i -E \
+-e '/CREATE ROLE "postgres";/d' \
+-e '/ALTER ROLE "postgres"/d' \
+-e 's/(NO)*SUPERUSER//g' \
+-e 's/(NO)*REPLICATION//g' \
+-e 's/(NO)*BYPASSRLS//g' \
+-e 's/GRANTED BY "[^"]*"//g' \
+roles.sql
+```
+
+<Highlight type="note">
+This command works only with the GNU implementation of sed (sometimes referred
+to as gsed). For the BSD implementation (the default on macOS), you need to
+add an extra argument to change the `-i` flag to `-i ''`.
+
+To check the sed version, you can use the command sed --version. While the GNU
+version will explicitly identify itself as GNU, the BSD version of sed
+generally doesn't provide a straightforward --version flag and will simply
+output an "illegal option" error.
+</Highlight>
+
+- `CREATE ROLE "postgres"`; and `ALTER ROLE "postgres"`: These statements are
+  removed because they require superuser access, which is not supported
+  by TimescaleDB Cloud.
+
+- `(NO)SUPERUSER` | `(NO)REPLICATION` | `(NO)BYPASSRLS`: These are permissions
+  that require superuser access.
+
+- `GRANTED BY role_specification`: The GRANTED BY clause can also have permissions that
+  require superuser access and should therefore be removed. Note: Per the
+  TimescaleDB documentation, the GRANTOR in the GRANTED BY clause must be the
+  current user, and this clause mainly serves the purpose of SQL compatibility.
+  Therefore, it's safe to remove it.
+
+Dump all plain tables and the TimescaleDB catalog from the source database:
+
+```
+pg_dump -d "$SOURCE" \
+  --format=plain \
+  --quote-all-identifiers \
+  --no-tablespaces \
+  --no-owner \
+  --no-privileges \
+  --file=dump.sql
+```
+
+a. `--no-tablespaces` is required because Timescale does not support
+   tablespaces other than the default. This is a limitation.
+b. `--no-owner` is required because tsdbadmin is not a superuser and cannot
+   assign ownership in all cases. This flag means that everything will be
+   owned by the tsdbadmin user in the target regardless of ownership in the
+   source. This is a limitation.
+c. `--no-privileges` is required because tsdbadmin is not a superuser and
+   cannot assign privileges in all cases. This flag means that privileges
+   assigned to other users will need to be reassigned in the target
+   database as a manual clean-up task. This is a limitation.
+
+Load the roles and schema into the target database.
+
+```
+psql -X -d "$TARGET" \
+  -v ON_ERROR_STOP=1 \
+  --echo-errors \
+  -f roles.sql \
+  -f dump.sql
+```
+
+TODO: Should we add a mention that customers can exclude the data from large
```

Comment:
@jgpruitt WDYT?

Reply:
yeah, that sounds like a good idea

Reply:
I think that the suggestion is generally correct, but I would suggest a very different approach, which includes something which is as yet missing in this text. I will push a revision with those suggestions.

Reply:
To clarify: IMHO we should suggest that people who are coming from plain postgres and who have large time-series tables convert those tables to hypertables. This means that they will exclude the data from those tables in their pg_dump, and backfill the tables into the hypertable.